[TensorExpr] Move IRSimplifier implementation to .cpp file.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35120 [NOT READY][TensorExpr] Bounds inference
* **#35169 [TensorExpr] Move IRSimplifier implementation to .cpp file.**

Currently it's implemented entirely in its header file, which causes
symbol redefinitions when included from several other fource files.

Differential Revision: [D20583661](https://our.internmc.facebook.com/intern/diff/D20583661)

Commit:
[TensorExpr] Move IRSimplifier implementation to .cpp file.

Currently it's implemented entirely in its header file, which causes
symbol redefinitions when included from several other fource files.

[ghstack-poisoned]

nickgg reviewed (changes requested):
Can we wait for https://github.com/pytorch/pytorch/pull/35127 please, since it rewrites that file.

Commit:
Update on "[TensorExpr] Move IRSimplifier implementation to .cpp file."


Currently it's implemented entirely in its header file, which causes
symbol redefinitions when included from several other fource files.

Differential Revision: [D20583661](https://our.internmc.facebook.com/intern/diff/D20583661)

[ghstack-poisoned]

Commit:
clang-format on "[TensorExpr] Move IRSimplifier implementation to .cpp file."


Currently it's implemented entirely in its header file, which causes
symbol redefinitions when included from several other fource files.

Differential Revision: [D20583661](https://our.internmc.facebook.com/intern/diff/D20583661)

[ghstack-poisoned]
